### ISSUE TITLE
Log file isn't being opened for triggers

### ISSUE THREAD

**Issue by pen-uj** [kauffman12/EQLogParser#206]
EQLP v2.2.46 (and 2.2.44), Windows 10.   Have had no trouble opening and monitoring EQ logs for DPS, spell casts, etc, but after adding triggers and a character (advanced tab) it does not pickup any triggers.  Also tried testing triggers - no joy.  Also, next to character name there are some oscillating red dots, when moused-over says "Attempting to open log", but it can't seem to open it, so triggers aren't detected.  

Note that the log is opened and monitored for DPS, etc from the main file menu with no problems.  This issue pre-dated me adding the specific character in the advanced tab, in that a trigger code shared by a friend was not picked up from my log file.

FWIW, GINA seems to open and monitor the log file for triggers just  fine.  Note: GINA is not running when I try EQLP for triggers.

Any thoughts why EQLP can't open a log file for triggers when it opens the same log files for fights, etc with no issues?

Thanks for any advice.

**Comment by pen-uj** [comment]
I checked the error log; nothing unusual at launch or after loading or monitoring a log file.  But as soon as I checked the box next to a character's name to activate triggers errors were logged.  I'll paste the annotated error log below:
------
2024-09-28 13:35:54,075 [1] INFO EQLogParser.App - EQLogParser: 2.2.46, DotNet: 8.0.7
2024-09-28 13:35:54,116 [1] INFO EQLogParser.MainWindow - Window Pos (61, 66)
2024-09-28 13:35:54,116 [1] INFO EQLogParser.MainWindow - Window Size (1289, 800)
2024-09-28 13:41:47,105 [1] INFO EQLogParser.MainWindow - Selected Log File: C:\Users\Public\Daybreak Game Company\Installed Games\EverQuest\Logs\eqlog_Sneakypetey_bristle.txt
2024-09-28 13:41:48,288 [1] INFO EQLogParser.MainWindow - Finished Loading Log File in 1 seconds.
**Note: at top of window it says "monitoring active -- " and lists the log file path
Note: trigger manager window is open, but no character has been checked as active yet.
Note: NOW clicked box next to sneakypetey (manage character);  red dots indicate "Attempting to open log file"**
2024-09-28 13:45:03,824 [1] INFO EQLogParser.AudioManager - Using audio device: Speakers (Cirrus Logic CS4206B (AB 92)).
2024-09-28 13:45:03,866 [1] ERROR EQLogParser.App - AppDispatcherUnhandledException
System.InvalidCastException: Specified cast is not valid.
   at System.Runtime.InteropServices.Marshal.ThrowExceptionForHR(Int32 errorCode)
   at WinRT.IObjectReference.As[T](Guid iid)
   at Windows.Media.SpeechSynthesis.SpeechSynthesizer.Make___objRef_global__Windows_Media_SpeechSynthesis_ISpeechSynthesizer2()
   at Windows.Media.SpeechSynthesis.SpeechSynthesizer.get__objRef_global__Windows_Media_SpeechSynthesis_ISpeechSynthesizer2()
   at Windows.Media.SpeechSynthesis.SpeechSynthesizer.get_Options()
   at EQLogParser.TriggerProcessor.SetVoiceRate(Int32 rate)
   at EQLogParser.TriggerProcessor..ctor(String id, String name, String playerName, String voice, Int32 voiceRate, String activeColor, String fontColor, Action`4 addTextEvent, Action`3 addTimerEvent)
   at EQLogParser.TriggerManager.<>c__DisplayClass42_0.<<HandleAdvancedConfig>b__4>d.MoveNext()
--- End of stack trace from previous location ---
   at EQLogParser.TriggerManager.HandleAdvancedConfig(TriggerConfig config)
   at EQLogParser.TriggerManager.ConfigDoUpdate(Object sender, EventArgs e)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by kauffman12** [comment]
It looks like the text to speech API is the problem. Not really opening the file.

What version of windows 10 is it? Like the OS build version. Is it behind in updates? Same goes for the sound drivers. That Cirrus Logic CS4206B doesn't seem too common. Do you have any USB or bluetooth headphones to try as an alternative sound device?

Also, can you check that some TTS voices are working in Windows? If you type 'Speech Settings' into the start menu search and bring up this UI I have below. Pick a voice and hit that preview voice button. Do the voices play OK for you?

![image](https://github.com/user-attachments/assets/e7c842cd-d583-4025-9c2b-c4bf92cceda6)

**Comment by pen-uj** [comment]
Thanks for the quick reply.

Answering your last question first, the TTS voices in Windows are working fine (Speech settings).

I wasn’t able to connect a USB headphone as an output device.

The drivers for that Cirrus Logic board are as up-to-date as they’ll ever be.

To be sure, this “PC” is an odd beast.  It’s an Intel Mac running Windows 10 (64bit) under Bootcamp, so the hardware is a circa 2013 27” iMac, booted into Windows (not emulated.) That said, every other PC software I’ve played with has run pretty well - as if I was running any older Windows 10 machine.

One odd thing is that GINA handles the text-to-speech fine.  I realize EQLP is both newer and clearly more sophisticated in most respects.   If I didn’t like EQLP so much for parsing I’d just use GINA.  Well, after playing EQ for literally decades I’ve finally starting hanging with a raiding guild, and they’re transitioning from GINA to EQLP, so the newest triggers they’re mainly supporting EQLP triggers.

Please let me know if there is a moderately easy fix - either at my end or in the EQLP code to recognize this particular sound board / driver.

Thanks for taking the time to look into this.

**Comment by kauffman12** [comment]
Can you run 'winver' and past what version it says?  It looks like the SpeakingRate option was add in like version 1709. If you do have an older version and can update that might be the best way to go. Otherwise, I can update EQLP to ignore this error and we can see if anything else breaks. Best case scenario is that everything works except for specifying a custom voice rate/speed.

**Comment by pen-uj** [comment]
Winver- Version 10.0 (build 10240)

I’ve been unable to get updates to stick, and had our IT guys try to figure out why it claims to run the updater but then it fails.


To make matters worse, (1) this windows machine is technically managed by my university.  They’d really wish I’d stop using such an old machine (I only use it for Bootcamp / Windows for EQ!) So I doubt I can get a newer driver for the sound.

Also, I installed Win10 to an external SSD (much faster and nicer), but that may have involved some slight-of-hand which the IT guys don’t like.

This is so convoluted I’d understand if you don’t want to get involved.  BUT if you’re willing to make a version of EQLP that ignores that error and see if that breaks anything else, I’d be willing to give it a shot.

I really appreciate your time on this (and on EQLP in general!!)

**Comment by kauffman12** [comment]
Alright I'll make a build you can try.

Also, I don't know how many ways you've tried to update Windows but you could download a slightly newer ISO and go through the normal setup process. It should give you an upgrade option. 

It's not too difficult to find a place to download old ISOs assuming you want to trust the links. Search on archive.org and there's a bunch to pick from. This search finds a bunch that would probably work.

https://archive.org/search?query=windows+10+1909

**Comment by pen-uj** [comment]
I believe installing from an iso has to go through the Bootcamp installer, which pretty much takes everything back to a blank slate and rebuilds.  As I understand it, the bootcamp installer has to build special drivers compatible with the non-PC hardware in the iMac.  I don’t think you can just do a major update from the iso, but I’ll double check.  As Apple no longer sells Intel-based Macs, bootcamp has been depreciated. Sadly, in the future I’ll just have to buy a windows PC to continue to run EQ.

In the meantime, please try a build that ignores that error, if it’s not a lot of trouble.  

Many thanks!

**Comment by kauffman12** [comment]
Try this version. I found a library I can use to change the voice speaking rates without using the windows API. See if you hear any popping or clicking sounds when changing the volume or voice rate options. Assuming things work.

https://github.com/kauffman12/EQLogParser/raw/master/Release/test/EQLogParser-install-2.2.46-test.exe

**Comment by pen-uj** [comment]
Thanks - Real life work calls and I won’t be able to test it out until tomorrow night, but will let you know as soon as I can.

**Comment by kauffman12** [comment]
Ok then just use the latest. 2.2.47. There's a couple other changes now that are worth keeping. No reason for the test build unless something else is wrong.

**Comment by pen-uj** [comment]
I had 5 minutes to check out 2.2.47.  It seemed to work, but I didn’t have time to mess around with it much.  Oddly, my UI settings in EQ were totally messed up - windows moved all around, as if it had been forced from windowed mode into full-screen and back.    I doubt that had anything to do with EQLP, tho.

More later, after I’ve had a chance to use it some.

Thanks again.

**Comment by pen-uj** [comment]
I checked out 2.2.47 some more and had no issues.  I was able to change the voice, and also the speaking rate just fine. Phrases typed into group chat triggered perfectly and overlays worked.  The only odd thing was I couldn’t get text typed into the trigger test window to trigger (after typing and hitting “Run Test”).

Nothing was reported in the error log.

I think this version solved the issues I was  having.  Thanks again for everything.
